src/gentropy/datasource/pan_ukbb_ld/ld.py

```diff
@@ -0,0 +1,214 @@
+"""Step to import filtered version of a LD matrix (block matrix)."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import hail as hl
+import numpy as np
+import pyspark.sql.functions as f
+from hail.linalg import BlockMatrix
+from pyspark.sql.window import Window
+
+from gentropy.common.session import Session
+from gentropy.config import PanUKBBConfig
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame, Row
+
+
+class PanUKBBLDMatrix:
+    """Toolset to work with Pan UKBB LD matrices."""
+
+    def __init__(
+        self,
+        pan_ukbb_ht_path: str = PanUKBBConfig().pan_ukbb_ht_path,
+        pan_ukbb_bm_path: str = PanUKBBConfig().pan_ukbb_bm_path,
+        ld_populations: list[str] = PanUKBBConfig().pan_ukbb_pops,
+        ukbb_annotation_path: str = PanUKBBConfig().ukbb_annotation_path,
+    ):
+        """Initialize.
+
+        Datasets are in hail native format.
+
+        Args:
+            pan_ukbb_ht_path (str): Path to hail table, source: gs://ukb-diverse-pops-public/ld_release
+            pan_ukbb_bm_path (str): Path to hail block matrix
+            ld_populations (list[str]): List of populations
+            ukbb_annotation_path (str): Path to pan-ukbb variant LD index with alleles flipped to match the order in OT variant annotation
+        Default values are set in PanUKBBConfig.
+        """
+        self.pan_ukbb_ht_path = pan_ukbb_ht_path
+        self.pan_ukbb_bm_path = pan_ukbb_bm_path
+        self.ld_populations = ld_populations
+        self.ukbb_annotation_output_path = ukbb_annotation_path
+
+    def align_ld_index_alleles(
```

Comment:
Where is this function used? I can't find any call.

Reply:
Or is this the part of that process you mentioned @addramir , that there's no step for it, but the data is generated?

Reply:
For this only needs to be run once for each Pan-UKBB ancestry, and then we store and use the resulting parquet DF for the other functions.